lib/system.nim

```diff
@@ -3124,5 +3124,12 @@ when not defined(createNimHcr) and not defined(nimscript):
 when notJSnotNims and not defined(nimSeqsV2):
```

Comment:
can something like this work:
```nim
# in lib/system.nim
when defined(nimdoc):
  proc prepareStrMutation*(s: var string) {.inline.} =
    ## String literals (e.g. "abc", etc) in the ARC/ORC mode are "copy on write",
    ## therefore you should call `prepareStrMutation` before modifying the strings
    ## via `addr`.
    runnableExamples("--gc:arc"):
      var x = "abc"
      var y = x
      prepareStrMutation(y)
      moveMem(addr y[0], addr x[0], 3)
      assert y == "abc"
elif notJSnotNims and not defined(nimSeqsV2):
  proc prepareStrMutation*(s: var string) {.inline.} = discard
```

```nim
# in lib/system/strs_v2.nim
proc prepareStrMutation*(s: var string) {.inline.} =
  {.cast(noSideEffect).}:
    let s = unsafeAddr s
    nimPrepareStrMutationV2(cast[ptr NimStringV2](s)[])
```

(see also https://github.com/nim-lang/RFCs/issues/309 which would simplify this)